app/models/http_proxy.rb

```diff
@@ -55,4 +55,11 @@ def nilify_empty_credentials
     self.username = nil if username.empty?
     self.password = nil if password.empty?
   end
+
+  def url_scheme
+    schemes = ['http://', 'https://']
+    unless schemes.any? { |scheme| url.start_with? scheme }
+      errors.add(:url, _("does not start with one of: %s") % schemes.join(', '))
+    end
+  end
```

Comment:
Really? :-) How about just `https?://` regexp. This could be even via `validates` without the need of an explicit validation method.